artifactory/utils/container/buildinfo.go

```diff
@@ -103,13 +112,18 @@ func setBuildProperties(buildName, buildNumber, project string, imageLayers []ut
 		return
 	}
 
-	// Skip if no properties were created
 	if len(props) == 0 {
 		log.Debug("Skipping setting properties - no properties created")
 		return nil
 	}
 
-	pathToFile, err := writeLayersToFile(imageLayers)
+	filteredLayers, err := filterLayersForVirtualRepository(imageLayers, serviceManager, originalRepo, repoDetails)
```

Comment:
there is a scenario where error is nil but the imageLayers slice is empty as well. So the function `writeLayersToFile(filteredLayers)` will be with empty slice.